tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/ScalaJSClassEmitter.scala

```diff
@@ -113,6 +117,39 @@ private[emitter] final class ScalaJSClassEmitter(
         }
         genES6Class(tree, allDefs)
     }
+
+    if (!tree.kind.isJSClass) {
+      entireClassDef
+    } else {
+      // Wrap the entire class def in an accessor function
+      import TreeDSL._
+      implicit val pos = tree.pos
+
+      val createClassValueVar =
+        envFieldDef("b", className, js.Undefined(), mutable = true)
+
+      val createAccessor = {
+        val classValueVar = envField("b", className)
+
+        val assignClassValueVar = js.Block(
+            entireClassDef,
+            classValueVar := envField("c", className)
+        )
```

Comment:
Could you move this inside the `js.If` or make it somehow clear that this is executed conditionally? It's kind of hard to read top-to-bottom otherwise.